edit/bzlmod/bzlmod.go

```diff
@@ -0,0 +1,320 @@
+/*
+Copyright 2023 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package bzlmod contains functions for working with MODULE.bazel files.
+
+package bzlmod
+
+import (
+	"strings"
+
+	"github.com/bazelbuild/buildtools/build"
+	"github.com/bazelbuild/buildtools/labels"
+)
+
+// Proxies returns the names of extension proxies (i.e. the names of variables to which the result
+// of a use_extension call is assigned) for the given extension with the given value of the
+// dev_dependency attribute.
+func Proxies(f *build.File, extBzlFile labels.Label, extName string, dev bool) []string {
```

Comment:
shouldn't this return just one string? especially given https://github.com/bazelbuild/bazel/pull/17920

Reply:
Clarified offline that:
```
ext1 = use_extension("//:foo.bzl", "foo")
ext1.tag()
ext2 = use_extension("//:foo.bzl", "foo")
ext2.tag()
```
is supported as Bazel returns the same proxy object in both cases.